src/core/js/models/lockingModel.js

```diff
@@ -0,0 +1,137 @@
+define([
+	'backbone'
+], function() {
+
+	var set = Backbone.Model.prototype.set;
+
+	_.extend(Backbone.Model.prototype, {
+
+		set: function(attrName, attrVal, options) {
+			var stopProcessing = !this.lockedAttributes || typeof attrName === "object" || typeof attrVal !== "boolean" || !this.isLocking(attrName);
+			if (stopProcessing) return set.apply(this, arguments);
+
+			var isSettingValueForSpecificPlugin = options && options.pluginName;
+			if (!isSettingValueForSpecificPlugin) {
+				console.error("Must supply a pluginName to change a locked attribute");
+				options.pluginName = "compatibility";
+			}
+
+			var pluginName  = options.pluginName;
+			var lockingValue = this.lockedAttributes[attrName] = !this.defaults[attrName];
+			var isAttemptingToLock = lockingValue === attrVal;
+
+			if (isAttemptingToLock) {
+
+				this.setLockValue(attrName, true, {pluginName:pluginName, skipcheck: true});
+
+				//console.log(options.pluginName, "locking", attrName, "on", this.get("_id"));
+				return set.call(this, attrName, lockingValue);
+
+			}
+
+			this.setLockValue(attrName, false, {pluginName:pluginName, skipcheck: true});
+
+			var totalLockValue = this.getLockValue(attrName, {skipcheck: true})
+			//console.log(options.pluginName, "attempting to unlock", attrName, "on", this.get("_id"), "lockValue", totalLockValue, this._lockedAttributesValues[attrName]);
+			if (totalLockValue === 0) {
+				//console.log(options.pluginName, "unlocking", attrName, "on", this.get("_id"));
+				return set.call(this, attrName, !lockingValue);
+			}
+
+			return this;
+
+		},
+
+		setLocking: function(attrName, defaultLockValue) {
+			if (this.isLocking(attrName)) return;
+			if (!this.lockedAttributes) this.lockedAttributes = {};
+			this.lockedAttributes[attrName] = defaultLockValue;
+		},
+
+		unsetLocking: function(attrName) {
+			if (!this.isLocking(attrName)) return;
+			if (!this.lockedAttributes) return;
+			delete this.lockedAttributes[attrName];
+			delete this._lockedAttributesValues[attrName];
+			if (_.keys(this.lockedAttributes).length === 0) {
+				delete this.lockedAttributes;
+				delete this._lockedAttributesValues;
+			}
+		},
+
+		isLocking: function(attrName) {
+			var isCheckingGeneralLockingState = attrName === undefined;
```

Comment:
sorry I know this is a bit picky, but I think these sorts of statements are more obvious if you do:
```var isCheckingGeneralLockingState = (attrName === undefined);```